controllers/helpers.go

```diff
@@ -1067,3 +1068,9 @@ func getCertificateFromFile(certificateFilePath string) ([]byte, error) {
 
 	return os.ReadFile(certificateFilePathTrimmed)
 }
+
+// ClusterPauseChangeAndInfrastructureReady is based on ClusterUnpausedAndInfrastructureReady, but
+// additionally accepts Cluster pause events.
+func ClusterPauseChangeAndInfrastructureReady(log logr.Logger) predicate.Funcs {
```

Comment:
I realize this is a trivial function, but should it have a unit test anyway?

Reply:
I see CAPI doesn't really have many specific tests for these, but I'll see what I can come up with.